src/client/app/components/view.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Store } from '@ngrx/store';
+import * as fromRoot from '../reducers';
+import { UI, Snippet } from '../actions';
+import { environment, applyTheme } from '../helpers';
+import { Strings } from '../strings';
+
+@Component({
+    selector: 'view',
+    template: `
+        <main [ngClass]="theme$|async">
+            <header class="command__bar">
+                <command class="title view-mode" [hidden]="isEmpty" icon="AppForOfficeLogo" [title]="snippet?.name"></command>
+            </header>
+            <editor readonly></editor>
+            <footer class="command__bar command__bar--condensed">
+                <command id="feedback" [title]="Feedback" icon="Emoji2" (click)="feedback()"></command>
+                <command icon="Color" [title]="theme$|async" (click)="changeTheme()"></command>
+            </footer>
+        </main>
+    `
+})
+
+
+export class ViewMode implements OnInit {
+    strings = Strings();
+    snippet: ISnippet;
+    showInfo: boolean;
+
+    constructor(
+        private _store: Store<fromRoot.State>,
+        private _route: ActivatedRoute
+    ) {
+        this._store.select(fromRoot.getCurrent).subscribe(snippet => {
+            this.snippet = snippet;
+        });
+    }
+
+    ngOnInit() {
+        let sub = this._route.params
+            .subscribe(async(params) => {
+                if (environment.current.host !== params.host.toUpperCase()) {
+                    environment.current.host = params.host.toUpperCase();
+                    await applyTheme(environment.current.host);
+                }
+
+                if (/private-samples/.test(location.hash)) {
```

Comment:
this._route.snapshot.url can get you the url segments without having to use the dependency on hash. Can you look into using this instead?